python/triton/compiler/compiler.py

```diff
@@ -272,6 +272,8 @@ def compile(src, target=None, options=None):
         ir.load_dialects(context)
         backend.load_dialects(context)
 
+    if os.environ.get("MLIR_DISABLE_MULTITHREADING", "0") == "1":
+        context.disable_multithreading()
```

Comment:
can this go in https://github.com/triton-lang/triton/blob/fe4fcdfcc95b2232e3011997ace414f88c9399c7/python/src/ir.cc#L1817 where we already handle a bunch of those env var

Reply:
ASAN ENV is here FWIW:
https://github.com/triton-lang/triton/blob/main/include/triton/Tools/Sys/GetEnv.hpp#L37

Reply:
right but I assumed it is because it has to be done after? Would be nice to move if possible as well, I don't think this is great to have this in the top function

Reply:
Sure, I don't feel strongly about it. If one ends up needing the other it probably makes sense to keep them together with a comment, that's all.